File: lambda_src/nodeJS/compute_pi.js
const AWS = require('aws-sdk');

exports.handler = async (event, context) => {
    //console.log('Received event:', JSON.stringify(event, null, 2));

    let body = {};
    const pi_count = 5000n;
    let statusCode = '200'
    const headers = {
        'Content-Type': 'application/json',
    };

    try {
        switch (event.httpMethod) {
            case 'DELETE':
                body = 'delete';
                break;
            case 'GET':
                body = {}
				var start = Date.now();
                //compute pi
                let i = 1n;
                let x = 3n * (10n ** pi_count);
                let pi = x;
                while (x > 0) {
                    x = x * i / ((i + 1n) * 4n);
                    pi += x / (i + 2n);
                    i += 2n;
                }
                body.pi = (pi / (10n ** 20n)).toString();
                //end
                var end = Date.now();
                body.time = end - start;
                break;
            case 'POST':
                body = 'post';
                break;
            case 'PUT':
                body = 'put';
                break;
            default:
                throw new Error(`Unsupported method "${event.httpMethod}"`);
        }
    } catch (err) {
        statusCode = '400';
        body = err.message;
    } finally {
        body = JSON.stringify(body);
    }

    return {
        statusCode,
        body,
        headers,
    };
};

File: lambda_src/python/compute_pi.py
import json
import time

def lambda_handler(event, context):
    body = {};
    pi_count = 5000
    statusCode = '200'
    headers = {
        'Content-Type': 'application/json',
    }

    try:
        body = {}
        start = time.perf_counter()
        # compute pi
        i = 1
        x = 3 * (10 ** pi_count)
        pi = x
        while (x > 0):
            x = x * i // ((i + 1) * 4)
            pi += x // (i + 2)
            i += 2
                
        body['pi'] = str(pi // (10 ** 20))
        end = time.perf_counter()
        function_time = end - start
        # to ms
        body['time'] = function_time * 1000

    except Exception as err:
        statusCode = '400'
        body = repr(err)
    
    return {
        'statusCode': statusCode,
        'body': json.dumps(body),
        'headers': headers
    }
